plugins/src/android/OptimoveInitProvider.java

```diff
@@ -95,4 +106,33 @@ public int update(@NonNull Uri uri, @Nullable ContentValues values, @Nullable St
         String value = resources.getString(resId);
         return TextUtils.isEmpty(value) ? null : value;
     }
+
+    private DeferredDeepLinkHandlerInterface getDDLHandler() {
+        return (Context context, DeferredDeepLinkHelper.DeepLinkResolution resolution, String link,
+                DeferredDeepLinkHelper.DeepLink data) -> {
+            try {
+                JSONObject dataJson = null;
+                if (null != data) {
+                    JSONObject deepLinkContent = new JSONObject();
+                    if (data.content != null) {
+                        deepLinkContent.put("title", data.content.title);
+                        deepLinkContent.put("description", data.content.description);
+                    }
+                    dataJson = new JSONObject();
+                    dataJson.put("data", data.data != null ? data.data : null);
+                    dataJson.put("content", deepLinkContent);
+                    dataJson.put("url", data.url);
+
+                }
+                JSONObject deepLink = new JSONObject();
+                deepLink.put("link", link);
+                deepLink.put("resolution", resolution.ordinal());
+                deepLink.put("data", dataJson);
+                OptimoveSDKPlugin.pendingDDL = deepLink;
```

Comment:
do we always save pendingDDL _and_ send message to js? If callback context was initialised, does it mean we call handler twice?

Reply:
I changed it to be like with the pending push